apps/full-site-editing/bin/js-unit-config.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Test configuration for the FSE plugin.
+ *
+ * Will match files such that:
+ *   1. Must be in the apps/full-site-editing/ directory
+ *   2. Must have .test.EXT at the end of the filename
+ *   3. EXT (above) must be one of js, ts, jsx, or tsx.
+ *
+ * Note: In order to use a different jest config for e2e tests, this config file
+ * must be kept in the bin/ folder to prevent it from being detected as the
+ * config file for e2e tests.
+ */
+
+/* eslint-disable import/no-extraneous-dependencies */
+
+// @wordpress/scripts manually adds additional Jest config ontop of
+// @wordpress/jest-preset-default so we pull in this file to extend it
+const defaults = require( '@wordpress/scripts/config/jest-unit.config.js' );
+const path = require( 'path' );
+
+// Basically, CWD, so 'apps/full-site-editing'.
+// Without this, it tries to use 'apps/full-site-editing/bin'
+const pluginRoot = path.resolve( './' );
+
+const config = {
+	...defaults,
+	rootDir: path.normalize( '../../../' ), // To detect wpp-calypso root node_modules
+	testMatch: [ `${ pluginRoot }/**/?(*.)test.[jt]s?(x)` ],
```

Comment:
A couple of things about the regex:

* `**/?` - could you elaborate on the reason to make the forward-slash optional?
* `(*.)` - any reason to have that in parens?
* `(*.)` - did you mean `(.*)`? Or does that have any other purpose?
* `.[jt]s?(x)` - did you mean to escape the dot?
* `[jt]s?(x)` - did you mean `[jt]sx?` Because with this one we'll allow `jx` and `tx` but not `js` and `ts`

Reply:
I think I was piecing together [different parts of this line in Gutenberg](https://github.com/WordPress/gutenberg/blob/9ac8c66ebc1f0a927093b3b1f27919ac6b3571c7/packages/scripts/config/jest-e2e.config.js#L13) and also [the default value in Jest](https://jestjs.io/docs/en/configuration.html#testmatch-arraystring) -- along with a poor understanding of regex and globs, I could easily mess something up here!

Confusingly, the existing regex appears to be matching the `.test.js` files in the correct directories (see the test-full-site-editing suite in the tests for this PR). 🤔 Looking at the docs again, it seems as if Jest uses `micromatch` for this config under the hood which might have different syntax than regex?

So the goal is to match any file ending with `.test.x` inside `apps/full-site-editing/full-site-editing-plugin` where x is one of `js`, `ts`, `jsx`, `tsx`. I made this runkit thing just to double check that it's matching/not matching how I expect. I think this syntax should work with micromatch :) https://runkit.com/noahtallen/5e41b4b68e170e0013b3d0b8

Reply:
As Noah pointed out the confusion here stems from `micromatch` that has a different syntax. You can check the relevant differences [here](https://github.com/micromatch/micromatch#extglobs), and after reading them the above expression should make sense. :)

Reply:
TIL! Thanks for elaborating 😉